spec.bs

```diff
@@ -252,19 +254,248 @@ dictionary AuctionAdConfig {
+      1. [=map/Set=] |bidGenerators|[|buyer|] to |perBuyerGenerator|.
+      1. TODO: add a perBiddingScriptUrlGenerator layer that replaces the list of IGs with a map
+        from biddingScriptUrl to a list of IGs.
+    1. Otherwise:
+      1. Let |perBuyerGenerator| be |bidGenerators|[|buyer|].
+      1. If |perBuyerGenerator| does not [=map/contain=] |signalsUrl|:
+        1. Let |perSignalsUrlGenerator| be a new [=per signals url bid generator=].
+        1. [=map/Set=] |perSignalsUrlGenerator|[|joiningOrigin|] to « |ig| ».
+        1. [=map/Set=] |perBuyerGenerator|[|signalsUrl|] to |perSignalsUrlGenerator|.
+      1. Otherwise:
+        1. Let |perSignalsUrlGenerator| be |perBuyerGenerator|[|signalsUrl|].
+        1. If |perSignalsUrlGenerator| does not [=map/contain=] |joiningOrigin|:
+          1. [=map/Set=] |perSignalsUrlGenerator|[|joiningOrigin|] to « |ig| ».
+        1. Otherwise:
+          1. [=list/Append=] |ig| to |perSignalsUrlGenerator|[|joiningOrigin|].
+1. [=map/For each=] |buyer| → |perBuyerGenerator| of |bidGenerators|:
+  1. [=map/For each=] |signalsUrl| -> |perSignalsUrlGenerator| of |perBuyerGenerator|:
+    1. TODO: Fetch |signalsUrl| synchronously.
+    1. [=map/For each=] |joiningOrigin| -> |groups| of |perSignalsUrlGenerator|:
+      1. [=list/For each=] |ig| of |groups|: 
+        1. Let |biddingScript| be the result of [=fetch bidding or scoring javascript=] with |ig|'s
+          [=interest group/bidding url=].
+        1. If |biddingScript| is an error, [=iteration/continue=].
+        1. TODO: [=Evaluate script=] with [=evaluate script/script=] set to |biddingScript|,
+          [=evaluate script/functionName=] set to "generateBid", and [=evaluate script/arguments=]
+          set to generateBid's arguemnts.
+
+</div>
+
+To <dfn>create a request</dfn> given a [=URL=] |url|, and a [=string=] <dfn for="create a request">
+|accept|</dfn>:
+1. Let |request| be a new [=request=] with the following properties:
+    :   [=request/URL=]
+    ::  |url|
+    :   [=request/header list=]
+    ::  A new [=header list=] containing a [=header=] named "`Accept`" whose value is |accept|
+    :   [=request/client=]
+    ::  `null`
+    :   [=request/window=]
+    ::  "`no-window`" TODO: verify
+    :   [=request/service-workers mode=]
+    ::  "`none`"
+    :   [=request/origin=]
+    ::  [=opaque origin=]
+    :   [=request/referrer=]
+    :: "`no-referrer`"
+    :   [=request/credentials mode=]
+    ::  "`omit`"
+    :   [=request/cache mode=]
+    ::  "`no-store`"
+    :   [=request/redirect mode=]
+    :: "`error`"
+1. Return |request|.
+
+To <dfn>fetch bidding or scoring javascript</dfn> with given [=URL=] |url|:
+1. Let |request| be the result of [=creating a request=] with |url|, and [=create a request/accept=]
+  set to "`application/javascript`".
+1. [=Fetch=] |request|.
+1. Let |response| be the result of [=fetch=] when it asynchronously completes.

```

Comment:
You'll actually need to use the [processResponseConsumeBody](https://fetch.spec.whatwg.org/#process-response-end-of-body) argument to Fetch to handle the response, as sketched in https://fetch.spec.whatwg.org/#fetch-elsewhere-fetch. You can write "wait for the processResponseConsumeBody algorithm to finish" or similar to avoid needing to convert your whole flow to callbacks.

I think it'd be fine to wait for a future PR to actually fix this up.

Reply:
done.